### PR TITLE
Complete removal of Python 3.8 support

### PR DESCRIPTION
**Pull request by Alexsandruss**
### Description

Remove Python 3.8 dependencies and other parts.

Changes:
- Update `sed` expression to correctly change sklearn version if python version condition is not present
- Change or remove requirements where Python 3.8 is mentioned
- Remove Python 3.8 from docs

---

Checklist to comply with before moving PR from draft:

**PR completeness and readability**

- [x] I have reviewed my changes thoroughly before submitting this pull request.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation to reflect the changes or created a separate PR with update and provided its number in the description, if necessary.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/intel/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [x] I have added a respective label(s) to PR if I have a permission for that.  
- [x] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [x] The unit tests pass successfully.
- [x] I have run it locally and tested the changes extensively.

**Performance**

- [x] I have measured performance for affected algorithms using [scikit-learn_bench](https://github.com/IntelPython/scikit-learn_bench) and provided at least summary table with measured data, if performance change is expected.
- [x] I have provided justification why performance has changed or why changes are not expected.

### CODE REVIEW

**Comment by ethanglaser** [comment]
LGTM and CI is good. Only other mention of 3.8 that could be removed is https://github.com/intel/scikit-learn-intelex/blob/main/daal4py/README.md#-pypi-channel - looks like this is very outdated but could fall within scope of this PR

**Review by ethanglaser — APPROVED** [review]
